File: src/BaseHmiTypes/Screens/Widgets/HmiScreenNavigationSettings.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Widgets;

public sealed class HmiScreenNavigationSettings
{
    public HmiProperty<string>? TargetScreen { get; set; }

    public string? ParameterFile { get; set; }

    public string? ParameterList { get; set; }

    public HmiDisplayParameterSource? ParameterSource { get; set; }

    public bool? UseVariableTarget { get; set; }

    public bool? PositionEnabled { get; set; }

    public bool? UseVariablePosition { get; set; }

    public HmiProperty<double>? PositionX { get; set; }

    public HmiProperty<double>? PositionY { get; set; }
}

File: typescript/base-hmi-types/src/screens/widgets/HmiScreenNavigationSettings.ts
import { HmiProperty } from "../base.js";
import { HmiDisplayParameterSource } from "./HmiDisplayParameterSource.js";

export class HmiScreenNavigationSettings {
  targetScreen?: HmiProperty<string>;
  parameterFile?: string;
  parameterList?: string;
  parameterSource?: HmiDisplayParameterSource;
  useVariableTarget?: boolean;
  positionEnabled?: boolean;
  useVariablePosition?: boolean;
  positionX?: HmiProperty<number>;
  positionY?: HmiProperty<number>;
}
